CMakeLists.txt

```diff
@@ -55,6 +55,20 @@ include(AddLLVM)
 include(AddMLIR)
 include(HandleLLVMOptions)
 
+# Check if the llvm SHA is consistent with llvm_version.txt
```

Comment:
Can you wrap the whole part inside "if(NOT IMEX_EXTERNAL_PROJECT_BUILD)"?
In tree build does not have "VCSRevision.h" and build will fail.
So it needs a different way of checking LLVM SHA. That support can be added in a separate PR once I figure out how to do it.

Reply:
> That support can be added in a separate PR once I figure out how to do it.

Can we just call git directly from llvm folder in this case?

Reply:
also, as we talked offline, in case of different llvm SHA, build_imex.py currently checks out llvm SHA and builds it, if this fails earlier than that, you need a way how to bypass this failure when someone uses build_imex.py.

Reply:
made the change